core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDocKey.java

```diff
@@ -56,6 +58,19 @@ public String getKey() {
 
     public void setKey(String key) {
         this.key = key;
+
+        int firstDotIndex = key.indexOf('.');
+        if (firstDotIndex == -1) {
+            this.topLevelGrouping = key;
+            return;
+        }
+        String withoutFirstDotIndex = key.substring(firstDotIndex + 1);
+        int secondDotIndex = withoutFirstDotIndex.indexOf('.');
+        if (secondDotIndex == -1) {
+            this.topLevelGrouping = withoutFirstDotIndex;
+            return;
+        }
+        this.topLevelGrouping = withoutFirstDotIndex.substring(0, secondDotIndex);
```

Comment:
We have this info here - https://github.com/quarkusio/quarkus/blob/c8b4f89b8fe4a097e698f2f18d52adc7ab3e1ebe/core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java#L58 with `configRootInfo.getName()`, maybe we can pass/set (possibly recursively) this from there instead of computing this?

Reply:
Sure, that makes sense. But I would really like to get this in for `1.3.1.Final` if the output is correct that is.
We can improve in a follow up when we don't have a release date breathing down our necks :)

Reply:
Ok to merge as is and I'll prepare a followup PR.

Reply:
Cool, thanks for checking.

If you can open a PR tomorrow, that would be great. Otherwise I might have time to do it